### PR TITLE
Migrate specialization auth handlers

### PR DESCRIPTION
**Pull request by mattthiessendev**
- refactor: migrate specialization auth handlers to its own directory
- refactor: SpecializationAuthorizatonHandler => LinkedToChatAuthorizationHandler

**Notes**
There are multiple ways to send change requests to our backend that relate to `Specialization` objects. This proposal is meant to open the namespace up for more descriptive `AuthorizationHandler`s related to `Specialization`s.

### CODE REVIEW

File: webapi/Auth/Specialization/LinkedToChatHandler.cs [hunk 1]
```diff
@@ -8,9 +8,9 @@
 namespace CopilotChat.WebApi.Auth;
```
Comment (mattthiessendev) [inline webapi/Auth/Specialization/LinkedToChatHandler.cs:8]: note: I did not change the namespace to look something like `ChopilotChat.WebApi.Auth.Specialization`. I didn't want to run into namespace collision with many sub namespaces having the name `Specialization`.

File: webapi/Auth/Specialization/LinkedToChatHandler.cs [hunk 2]
```diff
@@ -8,9 +8,9 @@
...
 
 /// <summary>
-/// Class implementing "authorization" that validates the user has access to a specialization.
+/// Class implementing "authorization" that validates the user has access to a specialization linked to the chat being targeted.
 /// </summary>
-public class SpecializationAuthorizationHandler(
+public class LinkedToChatHandler(
```
Comment (mattthiessendev) [inline webapi/Auth/Specialization/LinkedToChatHandler.cs:13]: note: I did not name this handler class to look something like `SpecializationLinkedToChatAuthorizationHandler`. I worried this would be too wordy and that the context surrounding the handler logic should be enough to let future readers understand what this is doing.

Thoughts?

Comment (mattthiessendev) [reply webapi/Auth/Specialization/LinkedToChatHandler.cs:13]: I am leaning towards `LinkedToChatAuthorizationHandler`

Comment (JTraill) [reply webapi/Auth/Specialization/LinkedToChatHandler.cs:13]: I like `LinkedToChatAuthorizationHandler`

Comment (mattthiessendev) [reply webapi/Auth/Specialization/LinkedToChatHandler.cs]: updated to `LinkedToChatAuthorizationHandler`

File: webapi/Auth/Specialization/LinkedToChatHandler.cs (hunk 1 above)
Comment (mattthiessendev) [reply webapi/Auth/Specialization/LinkedToChatHandler.cs]: it seems like having the namespace reflect the directory structure is a pretty common standard so I'll not break that for now since the only reason I would do is is aesthetic.

**Review by JTraill — APPROVED** [review]
